### PR TITLE
chore: add in-code documentation for ts form

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Fixes https://github.com/vaadin/flow/issues/8719

### CODE REVIEW

File: flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ const _validations = Symbol('validations');
 const _validating = Symbol('validating');
 const _validationRequestSymbol = Symbol('validationRequest');
 
+/**
+ * A Binder controls all aspects of a single form.
```
Comment (mshabarov) [inline flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts:28]: Don't we need to describe `T` and `M` parameters here?

File: flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts [hunk 2]
```diff
@@ -36,6 +41,13 @@ export class Binder<T, M extends AbstractModel<T>> extends BinderNode<T, M> {
 
   private [_validations]: Map<AbstractModel<any>, Map<Validator<any>, Promise<ReadonlyArray<ValueError<any>>>>> = new Map();
 
+  /**
+   *
```
Comment (mshabarov) [inline flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts:45]: Let's add some description on how the instance is created with this constructor. Weird to see params with no general description

Comment (haijian-vaadin) [reply flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts:48]: done

File: flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts (hunk 1 above)
Comment (haijian-vaadin) [reply flow-client/src/main/resources/META-INF/resources/frontend/form/Binder.ts:28]: done

**Pull request merged**